### PR TITLE
Actions daily workflows - only run in rubygems account

### PR DESCRIPTION
**Pull request by MSP-Greg**
# Description:

Support contributors by shutting off daily CI in forks.

- [x] Describe the problem / feature
- [ ] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by simi** [comment]
Same daily fail is in my fork as well. Is that problem with CI job? I see it pass in here.

**Comment by deivid-rodriguez** [comment]
Aha! I now understand the why of this request, I thought it was a contribution to help fight climate change :laughing:. It'd be interesting to understand why the daily bundler job fails on forks.

**Comment by deivid-rodriguez** [comment]
If I understand correctly, this only started happening very recently since nobody noticed this until now. Right?

**Comment by deivid-rodriguez** [comment]
Actually, this is not related to forks, but a real problem with HEAD versions in the canonical repo, since our daily workflow seems broken in the same way (and also with our daily notifications, since we didn't get notified about it).

So thanks for reporting this!

**Comment by simi** [comment]
22. 8. 2:20 was the first fail in my fork

**Comment by deivid-rodriguez** [comment]
Culprit is most likely https://github.com/ruby/ruby/commit/8a40e9b5a20314a509c8caaadce19a0b52680419.

**Comment by deivid-rodriguez** [comment]
The `set` library seems pure ruby so the easy fix is to vendor it. However, there's no rubygems.org release nor github repository. 

@hsbt Could you clarify what the situation regarding `set` as a default gem is? Also, could talk a bit about your criteria for promoting standard library code to default gems? When default gem promotion affects code heavily used by `bundler`, we need to do extra work to support it. Otherwise there's no benefit from the promotion, since `bundler` will load the version that comes with ruby and applications won't be able to upgrade.

**Comment by deivid-rodriguez** [comment]
Reported the issue at https://github.com/rubygems/rubygems/issues/3912 to reduce the noise on this PR.

**Comment by hsbt** [comment]
> @hsbt Could you clarify what the situation regarding `set` as a default gem is?

Because It's still preparation status.

File: .github/workflows/daily-bundler.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
 jobs:
   daily_bundler:
     runs-on: ubuntu-18.04
+    if: github.repository == "rubygems/rubygems"
```
Comment (eregon) [inline .github/workflows/daily-bundler.yml:10]: Can you also do this for `monthly-bundler.yml` ?

**Review by eregon — APPROVED** [review]
:+1: from me

File: .github/workflows/daily-bundler.yml (hunk 1 above)
Comment (MSP-Greg) [reply .github/workflows/daily-bundler.yml:10]: Fixed.  Sorry, I missed the monthly...

**Comment by MSP-Greg** [comment]
@deivid-rodriguez

> a contribution to help fight climate change

I didn't mention that.  Have you got any extra lights on?  How about that thermostat setting?

> since nobody noticed this until now

I normally turn actions off for repos with cron jobs.  Kind if a PITA.

**Review by deivid-rodriguez — APPROVED** [review]

**Comment by hsbt** [comment]
👍

**Pull request merged**